### DISCUSSION TITLE
every Version make Sketch bigger

### DISCUSSION THREAD

**Discussion by beabel** [espressif/arduino-esp32#10438] [Q&A]
Hello,

I have compiled my sketch (no Changes in Code) with Arduino IDE with different versions and each time it is bigger.
My problem is if it goes in this direction we end up needing more memory on the boards.

|Boardverwalter Version|Sketchsize|
| ------------- | ------------- |
|2.0.17|:white_check_mark: 1113105|
|3.0.0|:white_check_mark: 1288101|
|3.0.5|:x: 1312869|

Maximum on my Board is 1310720 Bytes

**Comment by me-no-dev** [comment]
Even though this is normal as more features are added, we are currently looking into how we could minimize the impact.

**Reply by horendus** [reply]
It would be great to have different build options to trim out unused features when size is a problem.

**Reply by Speedy2k** [reply]
It is really a big problem for me! Ihave undreds of device in the field that use OTA for firmware update that i can't upgrade to the new core because my program doesn't fit anymore with the updated arduino core!
